models/lsi.py

```diff
@@ -0,0 +1,96 @@
+from .abstract_model import AbstractModel
+from tools import create_address_books, create_dictionary_mids
+import numpy as np
+import pandas as pd
+from gensim import models, similarities
+from gensim.corpora import dictionary
+import os.path
+
+class LsiModel(AbstractModel):
+    def __init__(self, nb_recipients_to_predict=10):
+        self.nb_recipients_to_predict = nb_recipients_to_predict
+
+    def fit(self, training, training_info):
+        # store training sets
+        self.training = training
+        self.training_info = training_info
+        
+        print("creating train tokens")
+        train_tokens = training_info["tokens"].apply(lambda tokens: tokens.split(" ")).values.tolist() 
+        print("creating train dict")
+        train_my_dict = dictionary.Dictionary(train_tokens)
+        print("creating train corpus")
+        train_corpus = [train_my_dict.doc2bow(token) for token in train_tokens]
+        print("training Lsi model")
+        if os.path.isfile('../temp/model.lsi'): 
+            self.lsi = models.LsiModel.load('../temp/model.lsi')
+        else:
+            self.lsi = models.LsiModel(train_corpus, id2word=train_my_dict, num_topics=500)
+            self.lsi.save('../temp/model.lsi')
```

Comment:
There is an issue here. After i use this model once, for every other execution, it will always use the model trained the first time. Thus, if i use LsiModel during a cross-validation, the lsi model will fit for the first fold, and never fit again.

You should add a parameter, use_pretrained_model (False by default) to fix this issue